pkg/cloudprovider/providers/azure/azure_util_vmss.go

```diff
@@ -179,6 +181,11 @@ func (ss *scaleSet) getCachedVirtualMachine(nodeName string) (scaleSetVMInfo, er
 		return vm, nil
 	}
 
+	// Known node not managed by scale sets.
+	if ss.availabilitySetNodesCache.Has(nodeName) {
+		return scaleSetVMInfo{}, cloudprovider.InstanceNotFound
```

Comment:
why return `InstanceNotFound` if hit the cache?

Reply:
Hitting availabilitySetNodesCache means the instance is not managed by scale sets, so returns InstanceNotFound here. Callers of `getCachedVirtualMachine()` will also use this error message to identify whether the node existing or not.